identity_data/src/iosMain/kotlin/net/thechance/mena/identity/data/repository/location/MobileLocationRepositoryImpl.ios.kt

```
@@ -0,0 +1,39 @@
+package net.thechance.mena.identity.data.repository.location
+
+import dev.jordond.compass.geolocation.Geolocator
+import dev.jordond.compass.geolocation.GeolocatorResult
+import dev.jordond.compass.geolocation.MobileGeolocator
+import net.thechance.mena.identity.domain.entity.Coordinates
+import net.thechance.mena.identity.domain.exception.UnableToFindLocationException
+import net.thechance.mena.identity.domain.repository.MobileLocationRepository
+
+actual class MobileLocationRepositoryImpl( private val geocoder: GeocoderWrapper,
+    ) : MobileLocationRepository {
+
+    actual override suspend fun getCurrentLocation(): Coordinates? {
+        val geolocator: Geolocator = MobileGeolocator()
+        return when (val result = geolocator.current()) {
+            is GeolocatorResult.Error -> {
+                throw UnableToFindLocationException()
+            }
+
+            is GeolocatorResult.Success -> {
+                val location = result.data
+                Coordinates(
+                    latitude = location.coordinates.latitude,
+                    longitude = location.coordinates.longitude
+                )
+            }
+        }
+    }
+
+    actual override suspend fun getLocationName(
+        coordinates: Coordinates,
+    ): String {
+        val geocoder = geocoder.placeOrNull(coordinates)
+        return geocoder?.let {
+            listOfNotNull(it.subAdministrativeArea, it.administrativeArea, it.country)
+                .joinToString(", ")
+        }.orEmpty()
```

I think it won't be good to return empty address line, we can throw an exception here to force user to try picking a lockation again
I think you can ask on our channel to have better conversation with the team about the flow